### PR TITLE
feat: initial commit

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Migration of ocp-all-inclusive to TIM repository.

#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [x] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: tests/pr_test.go [hunk 1]
```diff
@@ -8,37 +7,65 @@ import (
...
 	assert.Nil(t, err, "This should not have errored")
 	assert.NotNil(t, output, "Expected some output")
 }
 
-func TestRunUpgradeExample(t *testing.T) {
+func TestRunCompleteExample(t *testing.T) {
+	t.Parallel()
+
+	// This test should always test the latest supported OCP version. Previous versions will be tested in other_test.go
+	versions := []string{"4.11"}
```
Comment (imprateeksh) [inline tests/pr_test.go:53]: If this check is only for the latest version , do we still need list and for loop to iterate over it?

**Review by imprateeksh — COMMENTED** [review]
Hi @Aashiq-J , 
Provided few comments, can you please check once?

**Review by vburckhardt — DISMISSED** [review]

File: main.tf [hunk 2]
```diff
@@ -1,3 +1,82 @@
...
+  # tflint-ignore: terraform_unused_declarations
+  logdna_validate_check     = regex("^${local.logdna_validate_msg}$", (!local.logdna_validate_condition ? local.logdna_validate_msg : ""))
+  sysdig_validate_condition = var.sysdig_instance_name != null && var.sysdig_access_key == null
+  sysdig_validate_msg       = "A value for var.sysdig_access_key must be passed when providing a value for var.sysdig_instance_name"
+  # tflint-ignore: terraform_unused_declarations
+  sysdig_validate_check = regex("^${local.sysdig_validate_msg}$", (!local.sysdig_validate_condition ? local.sysdig_validate_msg : ""))
+}
+
+module "observability_agents" {
+  # cluster-proxy required so observability images can be pulled from public registry
```
Comment (vburckhardt) [inline main.tf:67]: It is not relevant here - images can be pulled through private cse points

**Review by vburckhardt — COMMENTED** [review]
Thanks for the PR - just 2 comments.

**Pull request merged**